trpcultivate_phenotypes/src/Form/PhenoExperimentConfigurationForm.php

```diff
@@ -0,0 +1,339 @@
+<?php
+
+namespace Drupal\trpcultivate_phenotypes\Form;
+
+use Drupal\Core\Form\FormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Routing\RouteMatchInterface;
+use Drupal\tripal_chado\Database\ChadoConnection;
+use Drupal\trpcultivate_phenotypes\Service\TripalCultivatePhenotypesGenusOntologyService;
+use Drupal\trpcultivate_phenotypes\Service\TripalCultivatePhenotypesGenusProjectService;
+use Drupal\trpcultivate_phenotypes\Service\TripalCultivatePhenotypesTraitsService;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+/**
+ * Class definition of Experiment Configuration page.
+ */
+class PhenoExperimentConfigurationForm extends FormBase {
+
+  /**
+   * A Database query interface for querying Chado using Tripal DBX.
+   *
+   * @var \Drupal\tripal_chado\Database\ChadoConnection
+   */
+  protected ChadoConnection $chado_connection;
+
+  /**
+   * Genus-Ontotology service.
+   *
+   * @var \Drupal\trpcultivate_phenotypes\Service\TripalCultiavtePhenotypesGenusOntologyService
+   */
+  protected TripalCultivatePhenotypesGenusOntologyService $service_PhenoGenusOntology;
+
+  /**
+   * Genus-Project service.
+   *
+   * @var \Drupal\trpcultivate_phenotypes\Service\TripalCultivatePhenotypesGenusProjectService
+   */
+  protected TripalCultivatePhenotypesGenusProjectService $service_PhenoGenusProject;
+
+  /**
+   * Traits service.
+   *
+   * @var \Drupal\trpcultivate_phenotypes\Service\TripalCultivatePhenotypesTraitsService
+   */
+  protected TripalCultivatePhenotypesTraitsService $service_PhenoTraits;
+
+  /**
+   * Research experiment entity.
+   *
+   * @var \Drupal\tripal\Entity\TripalEntity
+   */
+  private $research_experiment;
```

Comment:
Add the type hint for all variables. There are a number of others missing the type hint.

```suggestion
  private TripalEntity $research_experiment;
```

Reply:
Implemented in commit https://github.com/TripalCultivate/TripalCultivate-Phenotypes/pull/189/commits/471929bfbfacb1ffa17bff80ade459d481aec398